lib/msf/core/constants.rb

```diff
@@ -49,6 +49,49 @@ module Msf
     ExcellentRanking => "excellent"
   }
 
+#
+# Stability traits
+#
+
+# Module does not crash the service
+CRASH_SAFE             = 'crash-safe'
+# Module crashes service, but service restarts.
+CRASH_SERVICE_RESTARTS = 'crash-service-restarts'
+# Module crashes service, and service remains down.
+CRASH_SERVICE_DEAD     = 'crash-service-dead'
+# Module crashes the OS, but OS restarts.
+CRASH_OS_RESTARTS      = 'crash-os-restarts'
+# Module crashes the OS, and OS remains down.
+CRASH_OS_DEAD          = 'crash-os-dead'
+# Module causes a resource (such as a file or data in database) to be unavailable for the service.
+SERVICE_RESOURCE_LOSS  = 'service-resource-loss'
+# Modules causes a resource (such as a file) to be unavailable for the OS.
+OS_RESOURCE_LOSS       = 'os-resource-loss'
+
+#
+# Side-effect traits
+#
+
+# Modules leaves payload or a dropper on the target machine
+PAYLOAD_ON_DISK        = 'payload-on-disk'
```

Comment:
This feels a little limiting. A payload that's left behind is an artifact. I'd still like to document **what** is left behind so I can clean it up later (the code should do this already).

Reply:
`ARTIFACTS_ON_DISK`, most likely. I think we have different approaches here, though. I'd prefer fewer constants and more notes on the **what** and **how**. But we both know these are consumables for the Platform before anything else.

Reply:
There is only so much meaning you can pack into a constant. If we want to be comprehensive, we'll want a lot of them. That'll make it more straightforward to integrate into the Platform. I'm fine with that approach so long as we're thorough.

Reply:
Ok yeah, sounds good. I'll do `ARTIFACTS_ON_DISK`.

We can definitely add more. Let me know what else you think we should add, we can do it.

Reply:
If we can decide on additional constants to add, `PAYLOAD_ON_DISK` may very well be okay. Up to you.